src/Contract/Operation/Reduceable.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+/**
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace loophp\collection\Contract\Operation;
+
+use loophp\collection\Contract\Collection;
+
+/**
+ * @template TKey
+ * @template T
+ */
+interface Reduceable
+{
+    /**
+     * Reduce a collection of items through a given callback.
+     *
+     * @param mixed $initial
+     *
+     * @return Collection<TKey, T>
+     */
+    public function reduce(callable $callback, $initial = null): Collection;
```

Comment:
This is not in sync with what's in the [operation](https://github.com/loophp/collection/pull/139/files?diff=split&w=1#diff-999d8f667f54a4a17936f748d9c02e4c5c9c727ef0181b9b714879554dd1ce16R31):
- missing type hint for the `callable`
- `$initial` should be the same type as the carry from the callback, right?

Reply:
It's done, but I'm very unsure about this. I have the feeling that we cannot really properly type this.